src/new-components/banner/banner.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import { Box } from '../box'
+import { Columns, Column } from '../columns'
+
+import styles from './banner.module.css'
+
+export type BannerTone = 'info' | 'promotion'
+
+type BannerProps = {
+    id?: string
+    /**
+     * The tone of the Banner. Affects the background color and the outline.
+     */
+    tone: BannerTone
+    /**
+     * The icon that should be added inside the Banner.
+     */
+    icon: React.ReactChild
+    /**
+     * The title to be displayed at the top of the Banner.
+     */
+    title: React.ReactNode
+    /**
+     * An optional description to be displayed inside the Banner.
+     */
+    description?: React.ReactNode
+    /**
+     * An optional action to displayed inside the Banner.
+     */
+    action?: React.ReactChild
+}
+
+const Banner = React.forwardRef<HTMLDivElement, BannerProps>(function Banner(
+    { id, tone, icon, title, description, action, ...props }: BannerProps,
+    ref,
+) {
+    return (
+        <Box
+            {...props}
+            ref={ref}
+            id={id}
+            role="banner"
+            aria-live="polite"
+            borderRadius="standard"
+            className={[styles.banner, styles[`banner-${tone}`]]}
+        >
+            <Columns space="medium" alignY="center">
+                <Column width="content" aria-hidden>
+                    {icon}
+                </Column>
+                <Column>
+                    <Box paddingY="xsmall">
+                        {description ? (
+                            <Box className={[styles.title, styles[`title-${tone}`]]}>{title}</Box>
```

Comment:
If you go with the `role="note"` suggestion, then I think we should make sure that the title of the banner becomes the semantic accessible name for the note element. To ensure this, please add a test that queries the element by role and accessible name:

```tsx
it('uses the title as the accessible name', () => {
  render(<Banner title="Hello world" />)
  expect(screen.getByRole('note', { name: 'Hello world' })).toBeInTheDocument()
})
```

If this does not work out of the box right now (which is most likely the case), then we may need to associate the title element to the container via `aria-labelledby`. Here's some pseudo-code:

```tsx
function Banner(props) {
  const titleId = useId() // import { useId } from 'react'
  return (
    <Box role="note" aria-labelledby={titleId} {…}>
      <Box id={titleId}>{title}</Box>
      // …
    </Box>
  )
}
```

Hope it helps.

Reply:
Oh, and to round things up, then the description (when present) should become the accessible description of the element. Using `aria-describedby` along the same lines.

Reply:
Indeed, it didn't workout _out of the box_, but then I put the appropriate props _in the Box_ and it worked! 😅

Thank you, all your notes were perfect. 👌